### PR TITLE
refactor link-to transform tests

### PR DESCRIPTION
**Pull request by chancancode**

### CODE REVIEW

File: packages/ember-template-compiler/tests/plugins/transform-link-to-test.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import TransfomrTestCase from '../utils/transform-test-case';
+import transformLinkTo from '../../lib/plugins/transform-link-to';
```
Comment (chancancode) [inline packages/ember-template-compiler/tests/plugins/transform-link-to-test.js]: this seems bad?

File: packages/ember-template-compiler/tests/utils/transform-test-case.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    let expected = deloc(ast(after, this.transforms));
+
+    this.assert.deepEqual(actual, expected);
+  }
+}
+
+function ast(template: string, plugins: ASTPluginBuilder[] = []): AST.Program {
+  let program: Option<AST.Program> = null;
+
+  function extractProgram(): ASTPlugin {
```
Comment (chancancode) [inline packages/ember-template-compiler/tests/utils/transform-test-case.ts]: hax

File: packages/ember-template-compiler/lib/plugins/transform-link-to.ts [hunk 3]
```diff
@@ -1,43 +1,38 @@
...
+      return b.mustache(content.path, content.params, content.hash, !escaped, loc);
 
     // The default case handles literals.
     default:
       return b.text(`${(content as AST.Literal).value}`, loc);
   }
 }
 
-function unsafeHtml(b: Builders, expr: AST.Expression) {
-  return b.sexpr('-html-safe', [expr]);
```
Comment (chancancode) [inline packages/ember-template-compiler/lib/plugins/transform-link-to.ts]: I'm not sure why we were doing this before. `{{{link-to foo ...}}}` -> `{{#link-to ...}}{{{foo}}}{{/link-to}}` seems easier and closer in spirit to what was written

File: packages/ember-template-compiler/tests/utils/transform-test-case.ts (hunk 2 above)
Comment (chancancode) [reply packages/ember-template-compiler/tests/utils/transform-test-case.ts]: is there a better way?

File: packages/ember-template-compiler/tests/plugins/transform-link-to-test.js (hunk 1 above)
Comment (chancancode) [reply packages/ember-template-compiler/tests/plugins/transform-link-to-test.js]: is there a better way?

Comment (wycats) [reply packages/ember-template-compiler/tests/plugins/transform-link-to-test.js]: TransfomrTestCase  -> TransformTestCase

**Review by rwjblue — APPROVED** [review]

**Comment by wycats** [comment]
r+

**Pull request merged**